packages/vincent-tools/vincent-tool-sdk/src/lib/lit-actions/vincent-policy-handler.ts

```diff
@@ -0,0 +1,65 @@
+import { z } from "zod"
+import { ethers } from "ethers"
+
+import { VincentPolicyDef } from "../types"
+import { getOnChainPolicyParams } from "../get-onchain-policy-params"
+import { formatZodError } from "../format-zod-error"
+
+declare const Lit: {
+    Actions: {
+        getRpcUrl: (args: { chain: string }) => Promise<string>;
+        setResponse: (response: { response: string }) => void;
+    }
+}
+declare const LitAuth: {
+    authSigAddress: string;
+}
+declare const toolParams: z.infer<VincentPolicyDef['toolParamsSchema']>
+
+const LIT_DATIL_VINCENT_ADDRESS = '0x78Cd1d270Ff12BA55e98BDff1f3646426E25D932';
+
+export const vincentPolicyHandler = ({ policyDef }: { policyDef: VincentPolicyDef }): (() => Promise<void>) => {
+    return async (): Promise<void> => {
+        try {
+            const parsedToolParams = parseToolParams(toolParams, policyDef.toolParamsSchema);
+
+            const onChainPolicyParams = await getOnChainPolicyParams({
+                yellowstoneRpcUrl: await Lit.Actions.getRpcUrl({
+                    chain: 'yellowstone',
+                }),
+                vincentContractAddress: LIT_DATIL_VINCENT_ADDRESS,
+                appDelegateeAddress: ethers.utils.getAddress(LitAuth.authSigAddress),
+                agentWalletPkpTokenId: parsedToolParams.userPkpTokenId,
+                toolIpfsCid: parsedToolParams.toolIpfsCid,
+                policyIpfsCid: policyDef.ipfsCid,
+                policyUserParamsSchema: policyDef.userParamsSchema,
+            });
+
+            const evaluateResult = await policyDef.evaluate({ toolParams: parsedToolParams, userParams: onChainPolicyParams });
+
+            Lit.Actions.setResponse({
+                response: JSON.stringify({
+                    ...evaluateResult,
+                    ipfsCid: policyDef.ipfsCid,
```

Comment:
Can we verify that the current LIT Action ipfsCid matches the policyDef ipfsCid?

Reply:
It would be best to just use `LitAuth.actionIpfsIds[0]` right? Use the actual IPFS CID that's being executed as reported by the Lit node